### DISCUSSION TITLE
Can I get top Searches in code?

### DISCUSSION THREAD

**Discussion by mikebhart** [algolia/instantsearch#4486] [General] [answered]
On my web page I want to show a list of all the top 5 searches made through algolia indicies.  Can someone point me to an example where I can access this data through code?

**Comment by Haroenv** [comment, marked as answer]
It can be found using the analytics API: https://www.algolia.com/doc/rest-api/analytics/#get-top-searches

**Comment by mikebhart** [comment]
Thanks for that I get {"status":401,"message":"The provided API key is missing the \"analytics\" ACL"}
does this mean I need to upgrade to premium?

code used 
curl -X GET -H "X-Algolia-API-Key: <they key>" -H "X-Algolia-Application-Id: <the app id> "https://analytics.algolia.com/2/searches?index=courses"

**Reply by Haroenv** [reply]
Either your account does not have access, you’re using a search api key or your app does not have access yes, don’t hesitate to reach out to support@algolia.com about these kind of things though
